File: C++/1260.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <bits/stdc++.h>
#define endl "\n" // don't use when you cover interactive problem
#define all(v) (v).begin(), (v).end()

using namespace std;
typedef long long ll;
typedef pair<int, int> pi;
typedef pair<ll, ll> pl;

int dfs_visited[1001];
int bfs_visited[1001];
vector<int> edge[1001];

void dfs(int n) {
    if (!dfs_visited[n]) {
        cout << n << ' ';
        dfs_visited[n] = 1;
        for (int i = 0; i < edge[n].size(); i++) {
            dfs(edge[n][i]);
        }
    }
    return;
}



int main() {
#ifndef ONLINE_JUDGE
    freopen("./input.txt", "r", stdin);
    freopen("./output.txt", "w", stdout);
#endif

    ios::sync_with_stdio(false);
    cin.tie(0), cout.tie(0);

    // 여기다가
    
    int N, M, V; cin >> N >> M >> V;
    for (int i = 0; i < M; i++) {
        int x, y; cin >> x >> y;
        edge[x].push_back(y);
        edge[y].push_back(x);
    }
    for (int i = 1; i <= N; i++) {
        sort(edge[i].begin(), edge[i].end());
    }
    dfs(V);
    cout << '\n';
    queue<int> bfs;
    bfs.push(V);
    bfs_visited[V] = 1;
    while (bfs.size()) {
        cout << bfs.front() << ' ';
        int f = bfs.front();
        vector<int> next = edge[f];
        for (int i = 0; i < next.size(); i++) {
            if (!bfs_visited[next[i]]) {
                bfs.push(next[i]);
                bfs_visited[next[i]] = 1;
            }
        }
        bfs.pop();
    }


    // 코드 작성하기
    return 0;
}

File: Python/1260.py
nodeNumber, edgeNumber, start = map(int, input().split())
edges = {x: [] for x in range(1, nodeNumber+1)}
for i in range(edgeNumber):
    edge = tuple(map(int, input().split()))
    edges[edge[0]].append(edge[1])
    edges[edge[1]].append(edge[0])

for key in edges:  # 정렬
    edges[key].sort()


DFS = [start]
BFS = [start]

# DFS
while True:
    L = len(DFS)
    for i in range(L-1, -1, -1):  # DFS 리스트의 끝부터 탐색해서
        node = DFS[i]
        for nxt in edges[node]:  # 각 노드에서 갈 수 있는 노드 중에
            if nxt not in DFS:  # 아직 안가본 곳이 있으면
                DFS.append(nxt)
                break
        else:  # 이 원소에는 더 갈곳이 없다
            continue
        break  # 이렇게 하면
    else:
        break  # 아하 여기가 종료 조건인가?
for x in DFS:
    print(x, end=' ')
print()

# BFS
iB = 1
while iB != 0:
    iB = 0  # 탐색하는 층의 노드의 개수
    for node in BFS[-iB:]:  # n-1 층
        for nxt in edges[node]:  # n층
            if nxt not in BFS:
                BFS.append(nxt)
                iB += 1
for x in BFS:
    print(x, end=' ')


"""아니 이게 왜 5퍼에서 틀림?

6 5 6
5 4
4 6
2 3
3 13 1 2
1 2
1 6"""